sameproject/sdk/same/dataset.py

```diff
@@ -0,0 +1,46 @@
+import yaml
+import os
+from urllib.parse import urlparse
+import pandas as pd
+
+def dataset(name, same_file='same.yaml'):
+    try:
+      env_var=os.environ['SAME_ENV'] if os.environ['SAME_ENV']!="" else "default"
+    except:
+      env_var="default"
+
+
+    with open(same_file, 'r') as file:
+        same_variables = yaml.safe_load(file)
+
+
+    urlpar = urlparse(same_variables['datasets'][name][env_var])
+    
+    url = 'https://gateway.ipfs.io/ipfs/'+urlpar.netloc+urlpar.path if urlpar.scheme=='ipfs' else same_variables['datasets'][name][env_var]
```

Comment:
split this out into a top level `if` like:
```
if parsed_url.scheme == "ipfs":
  [...]
```

This will make it easier to read. Add a comment above this line like "# If URL is an IPFS url, access it via the IPFS gateway"